src/app/consensus_test/consensus_test.ml

```diff
@@ -0,0 +1,229 @@
+open Core_kernel
+open Async
+
+module BlockFileOutput = struct
+  type t =
+    { height : int; parent_state_hash : string; previous_state_hash : string }
+  [@@deriving to_yojson]
+end
+
+type select_outcome = CandidateLonger | EqualLength | CandidateShorter
+
+module type CONTEXT = sig
+  val logger : Logger.t
+
+  val constraint_constants : Genesis_constants.Constraint_constants.t
+
+  val consensus_constants : Consensus.Constants.t
+end
+
+let context (logger : Logger.t) : (module CONTEXT) =
+  ( module struct
+    let logger = logger
+
+    let proof_level = Genesis_constants.Proof_level.None
+
+    let precomputed_values =
+      { (Lazy.force Precomputed_values.for_unit_tests) with proof_level }
```

Comment:
I would think we need be able to configure this via the CLI. The precomputed values we use for this test need to be the same as the precomputed values of the network that the blocks came from, otherwise the block selections will differ. Normally we load precomputed values from the config files. It might make sense to support specifying config files in this program, and to default to the unit test config when none are specified.

Reply:
That's a great point! Here is how I addressed configuring how we calculate `precomputed_values`:

Inside the `dune` file for this app, we specify that preprocessor dependency that's defined by an environment variable that defines a path on where the desired config lies. https://github.com/MinaProtocol/mina/pull/13737/files#diff-8d4396ff5b333f8ac32c0b3cd0cb75258cd3fe86b815e602f18527e30aa8e663R29

Additionally, we require a genesis ledger to compute `precomputed_values`. A CLI argument will specify this config file and expects a path to the config file.

So, when we compile the app, we define the compile config via an environment variable (`CONFIG_PATH`). Then when we run the app, we define a CLI argument that specifies the genesis ledger to be used.